test/build-change-graph-test.js

```diff
@@ -216,6 +216,60 @@ describe(buildChangeGraph, function() {
     ]));
   });
 
+  it('handles changed filenames with space', async function() {
```

Comment:
I just ran your test without your fix, and it still passes. Maybe push the test first to check that it is failing in CI?

Reply:
its passing, lol. 
so we encounter the bug when we do path.join and try to create directory from path.dirname

Reply:
Wouldn't you expect the response back to be `'"packages/package-a/sample index.js"'` before your fix?

Reply:
> Wouldn't you expect the response back to be `'"packages/package-a/sample index.js"'` before your fix?

yes, it only happening with uncommited changes :)

Reply:
In that case, I think you only want to modify the `dirtyChanges` so that the `union` works correctly.

Reply:
> In that case, I think you only want to modify the `dirtyChanges` so that the `union` works correctly.

updated